### DISCUSSION TITLE
Not able to setup turborepo in ubuntu 23.10

### DISCUSSION THREAD

**Discussion by ritikprajapat21** [vercel/turborepo#7960] [Help] [answered]
### Summary

I am trying to setup turborepo in ubuntu 23.10 on node version 20 and is not able to do it.
![Screenshot from 2024-04-15 15-03-40](https://github.com/vercel/turbo/assets/112960100/3d62ab58-b3f8-4f5a-8c06-eb7c77e25bf8)
As you can see in above image, I am stuck at selection of package manager, while I have installed all the package managers it still is showing not installed. I have also tried downgrading node version to 16 but it is still the same.

The command that I have tried for turborepo: 
![image](https://github.com/vercel/turbo/assets/112960100/a6b722ad-4618-4be4-a71f-2e31173d7343)

**Comment by anthonyshew** [comment]
This is erroring because you don't have any package managers installed. The error could be more graceful here - but you wouldn't be able to get too far with your repo without one. 😄

**Reply by ritikprajapat21** [reply]
@anthonyshew as I have said I have installed all the package managers but still showing this.

Any idea what might be causing this ??

**Reply by anthonyshew** [reply]
Are your package managers discoverable by other tooling? Did you do any customizations that would have moved the binaries from places where they would normally be installed? What does the system's `PATH` look like?

**Reply by chris-olszewski** [reply]
Can you provide the results of `which -a $PACKAGE_MANGER` for all the package managers you have installed? Also can you see what the results of the following commands are:
```
$ yarnpkg global bin
$ npm config get prefix
$ pnpm bin --global 
```

**Reply by ritikprajapat21** [reply]
Hi @chris-olszewski, Here is the ss of the commands output.
![image](https://github.com/vercel/turbo/assets/112960100/53a89420-4b2e-40ae-ace4-3536134121eb)

As you can see I have yarn and npm installed but it is still showing no package manager found.
![image](https://github.com/vercel/turbo/assets/112960100/04d1c608-a2bc-4218-b440-764e38cf64bd)

**Reply by ritikprajapat21** [reply]
Hi @anthonyshew, I have done some little bit of customization. 
Well here is the ss of the PATH variable:
![image](https://github.com/vercel/turbo/assets/112960100/f6d2ffce-2753-43da-8d6e-3ec946995db0)

**Reply by anthonyshew** [reply]
Odd that you're running the same commands we do and we're not picking those up. 🤔

https://github.com/vercel/turbo/blob/cc565e86be3506bc80592813a192217a8cc5b0c1/packages/turbo-utils/src/managers.ts#L44-L49

**Reply by ritikprajapat21** [reply]
@anthonyshew What could be done in this case??

**Reply by ayushagrwal** [reply]
Facing the same issue. @ritikprajapat21, were you able to resolve it?

**Reply by ayushagrwal** [reply]
@anthonyshew Pls suggest what could be done in this case? I'm using Ubuntu 22.04.4 LTS

**Comment by ritikprajapat21** [comment, marked as answer]
@ayushagrwal, I was not able to resolve it. But try to reinstall node.js again.

**Reply by ayushagrwal** [reply]
Yess it worked! Thanks man!
